src/sidebar/components/hooks/use-element-should-close.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { useEffect } = require('preact/hooks');
+
+const { listen } = require('../../util/dom');
+
+/**
+ * This hook adds appropriate `eventListener`s to the document when a target
+ * element (`closeableEl`) is open. Events such as `click` and `focus` on
+ * elements that fall outside of `closeableEl` in the document, or keypress
+ * events for the `esc` key, will invoke the provided `handleClose` function
+ * to indicate that `closeableEl` should be closed. This hook also performs
+ * cleanup to remove `eventListener`s when appropriate.
+ *
+ * @param {Object} closeableEl - Preact ref object:
```

Comment:
I honestly don't know how to document this param type here: is there a way to `typedef` preact objects?